src/lib/operators/table_scan.cpp

```diff
@@ -234,6 +234,38 @@ std::unique_ptr<AbstractTableScanImpl> TableScan::create_impl() const {
       return std::make_unique<ColumnVsColumnTableScanImpl>(input_table_left(), left_column_expression->column_id,
                                                            predicate_condition, right_column_expression->column_id);
     }
+
+    // Predicate pattern: <column> <binary predicate_condition> <uncorrelated subquery> (and vice versa)
```

Comment:
Comment += "This is faster than using the ExpressionEvaluator"?

Reply:
Isn't that the point of all the cases here?

Reply:
True, but you actually invoke the ExpressionEvaluator here, before handing the result over to another "evaluator", so that may raise an eyebrow.

Reply:
(would raise MY eyebrow and I would feel smart about shortening the code by removing this special case again, missing that I removed a perf-optimization for a very specific case)